contracts/AdapterStore.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+library MessengerTypes {
+    bytes32 public constant OFT_MESSENGER = bytes32("OFT_MESSENGER");
+    bytes32 public constant HYP_XERC20_ROUTER = bytes32("HYP_XERC20_ROUTER");
+}
+
+/**
+ * @dev A helper contract for chain adapters that support OFT or XERC20 messaging(via Hyperlane) on L1
+ * @dev Handles token => messenger/router mapping storage, as adapters are called via delegatecall and don't have relevant storage space
+ */
+contract AdapterStore is Ownable {
+    // (messengerType, l2ChainId, l1Token) => messenger address
```

Comment:
```suggestion
    // (messengerType, dstDomainId, l1Token) => messenger address
```

Reply:
Also, is the `l1Token` param always an L1 token? Could it be an L2 token when going from L2 to L1?